### PR TITLE
Sort & collapse kanji dictionary entries

### LINKED ISSUE #953: Ordering Kanji Dictionaries & Collapse doesn't work
I think this is a bug because we are unable to reorder kanji dictionaries like term dictionaries even if you change the priority number. 

Also can't collapse a kanji/hanzi dictionary. Some dictionaries like the Wiktionary one take up way too much space. 

Screenshot_20240516_224648_Kiwi Browser

よろしくお願いします

### LINKED ISSUE #1589: Dictionary ordering seems very unintuitive
The dictionary ordering ~~/priority~~ system right now seems very misleading, especially for someone who's just gotten into using the extension. 

- What does it mean for an accent dict to be of higher/lower ~~priority~~ ordering than a kanji dict or a normal(?) dict? In the end, the accent dicts get hoisted to the top and the kanji dicts are accessed separately. Unless i'm mistaken and there are dicts that don't just belong to one type (accent, kanji etc.), it doesn't make sense for them to be ordered together.

- The ~~priority~~ ordering doesn't seem to be working for kanji dicts for some reason. I've got `JPDB Kanji` at a higher priority than `KANJIDIC (English)`, but the latter appears before the former for some reason? Again, unless i'm mistaken and doing something wrong here

I understand that it's not everyday that one adds dicts and shuffles them around, but it's always a headache to go back to the list after a while and try to make sense of it 😅

Would be nice to have them grouped by type and then ordered within that type alone

Similar to #796, but focusing more on the unintuitive nature of ~~priorities~~ ordering as things stand and less on clutter

### PR DESCRIPTION
**Pull request by khaitruong922**
Fixes #953
Resolves #1589

### CODE REVIEW

**Comment by khaitruong922** [comment]
Screenshot
image
image

Current state of kanji dictionary collapse:
- Collapse button does not display fully on small width
image
- Collapsible max height may not be calculated correctly.
- Does not work correctly on `Collapsible` mode (only works when screen width is decreased then increased again)

**Comment by khaitruong922** [comment]
ready for review

File: ext/js/language/translator.js [hunk 1]
```diff
@@ -2022,6 +2030,31 @@ export class Translator {
...
+            // Sort by dictionary priority
+            let i = v2.dictionaryPriority - v1.dictionaryPriority;
+            if (i !== 0) { return i; }
+
+            // Sort by dictionary order
+            i = v1.dictionaryIndex - v2.dictionaryIndex;
+            if (i !== 0) { return i; }
+
+            // Sort by definition count
+            i = v2.definitions.length - v1.definitions.length;
```
Comment (MarvNC) [inline ext/js/language/translator.js:2052]: IMO this would cause confusion if it led to different sorting behavior on different entries

Comment (khaitruong922) [reply ext/js/language/translator.js]: I'll remove this since this sorting condition couldn't be reached.

**Review by MarvNC — APPROVED** [review]

**Pull request merged**